docs/guides/server/hostname2.adoc

```diff
@@ -0,0 +1,161 @@
+
+By explicitly setting the `--hostname`, we avoid a situation where tokens could be issued by a fraudulent issuer. In this context, the Keycloak server can be initiated using the following command:
+
+[source,bash]
+----
+kc.sh start --hostname my.keycloak.org
+----
+
+== Defining specific parts of the hostname option
+
+As demonstrated in the previous example, the scheme and port are not explicitly defined. In such cases, Keycloak automatically handles these aspects. For instance, the server would be accessible at https://my.keycloak.org:8443[https://my.keycloak.org:8443] based on the given example. However, a reverse proxy will typically expose Keycloak at the default ports, e.g. `443`. In that case it’s desirable to specify the full URL in the `--hostname` option rather than keeping the parts of the URL dynamic. The Keycloak server can then be started as:
+
+[source,bash]
+----
+kc.sh start --hostname https://my.keycloak.org
+----
+
+Similarly, your reverse proxy might expose Keycloak at a different context path. It is possible to configure Keycloak to reflect that via the `--hostname` and `--hostname-admin` options. See the following example:
+
+[source,bash]
+----
+kc.sh start --hostname https://my.keycloak.org:123/auth
+----
+
+== Utilizing Internal URL for communication among clients
+
+Keycloak has the capability to offer a separate URL for backchannel requests, enabling internal communication while maintaining the use of a public URL for frontchannel requests. Moreover, the backchannel is dynamically resolved based on incoming headers. Consider the following example:
+
+[source,bash]
+----
+kc.sh start --hostname https://my.keycloak.org --hostname-backchannel-dynamic true
+----
+
+In this manner, your applications, referred to as clients in Keycloak, can connect with Keycloak through your local network, while Keycloak itself remains publicly accessible at https://my.keycloak.org[https://my.keycloak.org].
+
+== Using edge TLS termination
+
+As you can observe, the HTTPS protocol is the default choice, adhering to Keycloak's commitment to security best practices. However, Keycloak also provides the flexibility for users to opt for HTTP if necessary. This can be achieved simply by specifying the HTTP listener, consult the https://www.keycloak.org/server/enabletls[TLS guide] for details. Assume you have your TLS-termination proxy operational. You can initiate the server as follows:
+
+[source,bash]
+----
+kc.sh start --hostname https://my.keycloak.org --http-enabled true
+----
+
+The result of this configuration is that you can continue to access Keycloak at https://my.keycloak.org[https://my.keycloak.org] via HTTPS, while the proxy interacts with the same Keycloak instance using HTTP and port `8080`.
```

Comment:
```suggestion
The result of this configuration is that you can continue to access {project_name} at https://my.keycloak.org[https://my.keycloak.org] via HTTPS, while the proxy interacts with the instance using HTTP and port `8080`.
```

If key material is not provided won't the server just provide http in this scenario?

Reply:
The newly added note could clarify it a bit. The examples assume the production mode, so we rely on that the key is provided.